app/models/upload.rb

```diff
@@ -23,12 +23,24 @@ def ok?
     ok
   end
 
+  def required
```

Comment:
These two methods are basically view data.  We should try to keep our views/presentation logic separate from our models.  These are prime candidates for helper methods.

You can create a `helpers/dashboards_helper.rb` and define these methods there (the methods would receive the `upload` object as a parameter)

Reply:
Moved and rspeced